shared/app/nav.native.js

```diff
@@ -185,25 +184,17 @@ class MainNavStack extends Component<any, any> {
     this.setState({verticalOffset: frameData.height - (isIPhoneX ? 45 : 20)})
   }
 
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const {routeSelected, routeStack} = nextProps
-    if (tabIsCached.hasOwnProperty(routeSelected)) {
```

Comment:
my understanding is we keep a subset of key/values around always. Instead we can just do this in the render function as it never triggers anything so it doesn't need to be in state